contracts/HubPool.sol

```diff
@@ -144,67 +158,73 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
-        address destinationToken,
-        uint256 destinationChainId
+        address destinationToken
     ) public onlyOwner {
         whitelistedRoutes[originToken][destinationChainId] = destinationToken;
-
-        emit WhitelistRoute(originToken, destinationChainId, destinationToken);
+        emit WhitelistRoute(destinationChainId, originToken, destinationToken);
     }
 
-    // TODO: the two functions below should be called by the Admin contract.
-    function enableL1TokenForLiquidityProvision(address l1Token) public onlyOwner {
+    function enableL1TokenForLiquidityProvision(address l1Token, bool isWeth) public onlyOwner {
         // NOTE: if we run out of bytecode this logic could be refactored into a custom token factory that does the
         // appends and permission setting.
-        ExpandedERC20 lpToken = new ExpandedERC20(
-            _append("Across ", IERC20Metadata(l1Token).name(), " LP Token"), // LP Token Name
-            _append("Av2-", IERC20Metadata(l1Token).symbol(), "-LP"), // LP Token Symbol
-            IERC20Metadata(l1Token).decimals() // LP Token Decimals
-        );
-        lpToken.addMember(1, address(this)); // Set this contract as the LP Token's minter.
-        lpToken.addMember(2, address(this)); // Set this contract as the LP Token's burner.
-        lpTokens[l1Token] = LPToken({ lpToken: lpToken, isEnabled: true });
+        if (pooledTokens[l1Token].lpToken == address(0))
+            pooledTokens[l1Token].lpToken = lpTokenFactory.createLpToken(l1Token);
+
+        pooledTokens[l1Token].isEnabled = true;
+        pooledTokens[l1Token].isWeth = isWeth;
+        pooledTokens[l1Token].lastLpFeeUpdate = uint32(getCurrentTime());
+
+        emit L1TokenEnabledForLiquidityProvision(l1Token, isWeth, pooledTokens[l1Token].lpToken);
     }
 
     function disableL1TokenForLiquidityProvision(address l1Token) public onlyOwner {
-        lpTokens[l1Token].isEnabled = false;
+        pooledTokens[l1Token].isEnabled = false;
+        emit L2TokenDisabledForLiquidityProvision(l1Token, pooledTokens[l1Token].isWeth, pooledTokens[l1Token].lpToken);
     }
 
-    // TODO: implement this. this will likely go into a separate Admin contract that contains all the L1->L2 Admin logic.
-    // function setTokenToAcceptDeposits(address token) public {}
-
     /*************************************************
      *          LIQUIDITY PROVIDER FUNCTIONS         *
      *************************************************/
 
     function addLiquidity(address l1Token, uint256 l1TokenAmount) public payable {
-        require(lpTokens[l1Token].isEnabled, "Token not enabled");
+        require(pooledTokens[l1Token].isEnabled, "Token not enabled");
         // If this is the weth pool and the caller sends msg.value then the msg.value must match the l1TokenAmount.
         // Else, msg.value must be set to 0.
-        require((address(l1Token) == address(l1Weth) && msg.value == l1TokenAmount) || msg.value == 0, "Bad msg.value");
```

Comment:
this part changed as I now store the `isWeth` within the `pooledTokens`. This makes things a bit cleaner and we dont need an extra state variable.

Reply:
what's the gas savings of one state variable `address` vs 1 `bool` per `PooledToken` though?

Reply:
ye, you right...it might be cheaper to have one state variable vs the extra bool. I think the cost difference is pretty small so overall we should optimise for the easier to reason about path.